### ISSUE TITLE
$state lazy-load controller

### ISSUE THREAD

**Issue by matejkramny** [angular-ui/ui-router#1002]
This is a feature which could save a lot of overhead when first loading the app, because it only loads controllers when they are actually required.

I use require.js in the project.

What i've tried:
- Using resolve: { } to load the controller, then fire deferred.resolve, however angular doesn't find the controller
- Using controllerProvider, however the called function has to return a name of the controller not load it dynamically.

Way I could imagine this working..
I define my state

```
$stateProvider.state('mystate', {
  activated: function($q) {
    var deferred = $q.defer();
    require(['myController'], function(controller) {
      deferred.resolve();
    });
    return deferred.promise;
  },
  controller: 'myController'
});
```

the `Activated` function gets called when the $state is being activated to, but angular ui waits before starting to load the state/controller.

Question.. Could this work?
Is there something existing that could work right now?

Notes:
I have no idea how angular-ui-router works internally

Thanks in advance.
